scripts/morty/analysis.py

```diff
@@ -0,0 +1,273 @@
+
+        Returns
+        -------
+        scaling_factor : float
+            Factor to scale time in (s) by
+        xlab : str
+            Time label to use
+        """
+        tf = self.run_params['final_time']
+        second = 1
+        millisecond = 1e-3 * second
+        minute = 60 * second
+        hour = 60 * minute
+        day = 24 * hour
+        year = 365.25 * day
+        time_list = [year, day, hour, minute, second, millisecond]
+        label_list = ['yr', 'd', 'hr', 'min', 's', 'ms']
+        for ti, time in enumerate(time_list):
+            if time > tf:
+                continue
+            xlab = f'Time [{label_list[ti]}]'
+            scaling_factor = 1 / time
+            return scaling_factor, xlab
+
+    def _method_change(
+            self,
+            methods,
+            method_name,
+            xfactor=1,
+            methodname_extension=''):
+        """
+        Used to handle concentration plotting comparisons with a single
+            method change.
+
+        Parameters
+        ----------
+        methods : list
+            List of methods to apply
+        method_name : str
+            Key for dictionary to apply method to `run_params`
+        xfactor : float
+            Value to scale `x_vals` by
+        methodname_extension : str (optional)
+            String to add to method name in plotting
+
+        Returns
+        -------
+        xs : list of list
+            Contains the `x` values to be plotted
+        ys : list of list
+            Contains the `y` values to be plotted
+        labs : list of list
+            Contains the ordered labels to use in plotting
+
+
+        """
+        xs, ys, labs = [], [], []
+        self.parasitic_results = []
+        for method in methods:
+            self.run_params[method_name] = method
+            self.data_params = DataHandler(self.run_params).data_params
+            solver = solvers.DiffEqSolvers(
+                self.run_params, self.data_params, run=False)
+            x_vals = solver.run_params['times']
+            xs.append(xfactor * x_vals)
+            result_matrix = solvers.DiffEqSolvers(
+                self.run_params, self.data_params).result_mat
+            ys.append(result_matrix)
+            labs.append(f'{method}{methodname_extension}')
+            self.parasitic_results.append(self._parasitic_abs(result_matrix))
+        return xs, ys, labs
+
+    def _save_data(self, xs, ys, labs, xlab, ylab, savename):
+        """
+        Saves the collected data into a single dictionary
+
+        Parameters
+        ----------
+        xs : list of list
+            Contains the `x` values to be plotted
+        ys : list of list
+            Contains the `y` values to be plotted
+        labs : list of list
+            Contains the ordered labels to use in plotting
+        xlab : str
+            Name of xlabel
+        ylab : str
+            Name of ylabel
+        savename : str
+            Name of file to save as
```

Comment:
best to clarify whether the filetype extension is expected and, if so, what filetypes are right...